### PR TITLE
Add mypy & pylint check to the build scripts

### PR DESCRIPTION
**Pull request by ilappe**
## Status
- [x] In Progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/40912

### CODE REVIEW

**Comment by Itay4** [comment]
using `# type: ignore` is bad practice, better to ignore specific error codes
also, why do we ignore instead of fixing ?
last, note the new lgtm alert

File: Tests/configure_and_test_integration_instances.py [hunk 1]
```diff
@@ -923,10 +925,9 @@ def get_tests(build: Build) -> List[str]:
             # skip test button testing
             logging.debug('Not running instance tests in nightly flow')
             tests_for_iteration = []
-        elif filtered_tests:
-            tests_for_iteration = [test for test in tests if test.get('playbookID', '') in filtered_tests]
         else:
-            tests_for_iteration = tests
+            tests_for_iteration = [test for test in tests
+                                   if not filtered_tests or test.get('playbookID', '') in filtered_tests]
```
[the remark this answers is not in the document]
Comment (ilappe) [reply Tests/configure_and_test_integration_instances.py:930]: because of
error: Incompatible types in assignment

**Comment by ilappe** [comment]
@yaakovi please wait with the re-review
i need something small to fix

**Review by yaakovi — DISMISSED** [review]

**Comment by yaakovi** [comment]
> @yaakovi please wait with the re-review I need something small to fix

missed that comment...

**Review by yaakovi — APPROVED** [review]

**Pull request merged**